### PR TITLE
Create screen for browsing competition

### PR DESCRIPTION
**Pull request by kubaolszewski**

### CODE REVIEW

File: lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+              color: Colors.white,
+            )),
+        title: Text(
+          context.localizations.searchHeader,
+          style: CommonTextStyles.basicWhiteText,
+        ),
+      ),
+      body: BlocProvider<CompetitionCubit>(
+        create: (context) => competitionCubit..searchingLeaguesByName(nameQuery),
+        child: BlocBuilder<CompetitionCubit, CompetitionState>(
```
Comment (MBanaszynski) [inline lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart]: Move bloc builder to separated file `searched_leauges_view`

File: lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+              ),
+            );
+          },
+        ),
+      ),
+    );
+  }
+}
+
+final competitionCubit = Modular.get<CompetitionCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart]: When you use this cubit only once, please don't create parameters for them. In provider use `Modular.get<>()`

File: lib/app/features/competiton/searched_competition_screens/searched_teams/searched_teams_screen.dart [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              color: Colors.white,
+            )),
+        title: Text(
+          context.localizations.searchHeader,
+          style: CommonTextStyles.basicWhiteText,
+        ),
+      ),
+      body: BlocProvider<CompetitionCubit>(
+        create: (context) => competitionCubit..searchingTeamsByName(nameQuery),
+        child: BlocBuilder<CompetitionCubit, CompetitionState>(
```
Comment (MBanaszynski) [inline lib/app/features/competiton/searched_competition_screens/searched_teams/searched_teams_screen.dart]: Same here

File: lib/app/features/landing/landing.dart [hunk 4]
```diff
@@ -36,7 +38,6 @@ void dispose() {
 class _LandingScreenState extends State<LandingScreen> {
   @override
   Widget build(BuildContext context) {
-    final landingCubit = Modular.get<LandingCubit>();
     return BlocProvider(
       create: (context) => landingCubit,
```
Comment (MBanaszynski) [inline lib/app/features/landing/landing.dart]:
```suggestion
      create: (context) => Modular.get<LandingCubit>(),
```

File: lib/app/features/landing/landing.dart [hunk 5]
```diff
@@ -15,6 +15,8 @@ class LandingScreen extends StatefulWidget {
   State<LandingScreen> createState() => _LandingScreenState();
 }
 
+final landingCubit = Modular.get<LandingCubit>();
```
Comment (MBanaszynski) [inline lib/app/features/landing/landing.dart]:
```suggestion
```

File: lib/models/league_model/league_model.dart [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        League? league,
+        Country? country,
+        List<Season>? seasons,
+    }) = _LeagueModel;
+
+    factory LeagueModel.fromJson(Map<String, dynamic> json) => _$LeagueModelFromJson(json);
+}
+
+@freezed
+class Country with _$Country {
```
Comment (MBanaszynski) [inline lib/models/league_model/league_model.dart]: Please move the rest of classes to separated files

File: lib/models/team_model/team_model.dart [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        String? country,
+        int? founded,
+        bool? national,
+        String? logo,
+    }) = _Team;
+
+    factory Team.fromJson(Map<String, dynamic> json) => _$TeamFromJson(json);
+}
+
+@freezed
```
Comment (MBanaszynski) [inline lib/models/team_model/team_model.dart]: Same here

**Review by MBanaszynski — CHANGES_REQUESTED** [review]

File: lib/app/features/landing/landing.dart (hunk 5 above)
Comment (kubaolszewski) [reply lib/app/features/landing/landing.dart]: done

File: lib/app/features/landing/landing.dart (hunk 4 above)
Comment (kubaolszewski) [reply lib/app/features/landing/landing.dart]: done

File: lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart (hunk 2 above)
Comment (kubaolszewski) [reply lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart]: done

File: lib/app/features/competiton/searched_competition_screens/searched_teams/searched_teams_screen.dart (hunk 3 above)
Comment (kubaolszewski) [reply lib/app/features/competiton/searched_competition_screens/searched_teams/searched_teams_screen.dart]: done

File: lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart (hunk 1 above)
Comment (kubaolszewski) [reply lib/app/features/competiton/searched_competition_screens/searched_leagues/searched_leagues_screen.dart]: done

File: lib/models/league_model/league_model.dart (hunk 6 above)
Comment (kubaolszewski) [reply lib/models/league_model/league_model.dart]: done

File: lib/models/team_model/team_model.dart (hunk 7 above)
Comment (kubaolszewski) [reply lib/models/team_model/team_model.dart]: done

**Review by MBanaszynski — APPROVED** [review]

**Pull request merged**